pmd-core/src/main/java/net/sourceforge/pmd/PmdAnalysis.java

```diff
@@ -287,7 +288,14 @@ void performAnalysisImpl(List<? extends GlobalReportBuilderListener> extraListen
         GlobalAnalysisListener listener;
         try {
             @SuppressWarnings("PMD.CloseResource") AnalysisCacheListener cacheListener = new AnalysisCacheListener(configuration.getAnalysisCache(), rulesets, configuration.getClassLoader());
-            listener = GlobalAnalysisListener.tee(listOf(createComposedRendererListener(renderers), GlobalAnalysisListener.tee(listeners), GlobalAnalysisListener.tee(extraListeners), cacheListener));
+            if (configuration.isProgressBar()) {
```

Comment:
should we try to be smart and have this off by default when the terminal is not interactive (ie: output is being redirected to a file)?

In Java this can be done checking if `System.console()` is `null`, but this is a proxy, as it fails if either the STDOUT or STDIN are not connected to a device… checking solely for STDOUT would require an equivalent to C's [`isaaty`](https://www.man7.org/linux/man-pages/man3/isatty.3.html), which is unfortunately not available without JNI.

Reply:
Should this maybe be done inside the shell wrapper instead? https://stackoverflow.com/questions/911168/how-can-i-detect-if-my-shell-script-is-running-through-a-pipe

Reply:
maybe… but `PmdAnalysis` is a programatic interface, if someone is using it embedded somewhere else (ie: IDEs) it would be up to them to replicate this / add the flag to retain the previous behavior.

Reply:
You're right... So maybe progress should be disabled by default when creating a `PMDConfiguration` programmatically? (although for the CLI interface it stays enabled by default)